merge bayercontour* meters into bayercontournext

krystophv opened the pull request:
Redux reducer modifications to take any existing `bayercontour*` device selections and collapse them down to `bayercontournext`.

krystophv commented:
![](https://github.trello.services/images/mini-trello-icon.png) [[uploader] Collapse Bayer meters into a single "Bayer Contour" selection / prevent "Unknown Device"](https://trello.com/c/06aRAIAq/237-uploader-collapse-bayer-meters-into-a-single-bayer-contour-selection-prevent-unknown-device)

gniezen commented on app/reducers/users.js:
```diff
@@ -156,7 +156,15 @@ export function targetDevices(state = {}, action) {
+          let targetDevices = _.get(membership, ['profile', 'patient', 'targetDevices'], []);
+          // collapse all bayercontour devices into bayercontournext
+          targetDevices = _.uniq(_.map(targetDevices, function(device) {
+            if (device.startsWith('bayercontour')) {
```
While this will work for now, there will be a separate driver (and entry) for the Bayer Contour meter, see https://github.com/tidepool-org/chrome-uploader/pull/168/files. Probably better to match on `bayercontournextusb`, `bayercontourusb`, `bayercontournextlink` and `bayercontournext`. We'll also need to merge `abbottfreestylefreedomlite` into `abbottfreestylelite`.

krystophv commented:
Ok, added the abbottfreestylefreedomlite translation, made the BCN check a little more specific and added a test to make sure that `bayercontour` doesn't get munged by the reducer (for when we add that device in).

gniezen commented:
Looks great, thanks!
